### ISSUE TITLE
Use of the target in the base model

### ISSUE THREAD

**Issue by RX-0-95** [vturrisi/solo-learn#91]
Hello, greate work in those models. I have some question in the implementation of those models. In the base model of all self-supervised model, there is a  [_shared_step function](https://github.com/vturrisi/solo-learn/blob/main/solo/methods/base.py#:~:text=def%20_shared_step(self,1%20and%20acc) used to forward calculate the 'class_loss' of the network. And the 'class loss' is added to the other constrastive loss later to backprob and update the network. The 'class_loss' is compute use label from the data and the cross entropy function, but in self-supervised learning, we are not allow to use label during the pretrain. Maybe I missunderstand your implementation, if this is the case, please let me know where did I make the mistake.

**Comment by vturrisi** [comment]
Hey,
Thanks for the message. If you check the `_shared_step(...)` function, you will see that `_base_forward(...)` is called. Here (https://github.com/vturrisi/solo-learn/blob/main/solo/methods/base.py#L309) you can see that we detach the features when forwarding through the classifier. So when we sum the losses, we are binding two computational graphs into a single one (the classification loss that only gives gradients to the classifier) and the self-sup loss (that gives gradients to all module but the classifier).
